From 935afce06e23e9a0277d46c7746482ff37bbe8ae Mon Sep 17 00:00:00 2001
From: Srinivas Neeli <srinivas.neeli@xilinx.com>
Date: Thu, 6 Aug 2020 13:38:11 +0530
Subject: [PATCH 1500/1851] gpio: gpio-xilinx: Check return value of
 of_property_read_u32

commit cf1eebd8716c5260bbe068710c3a26bcdee56343 from
https://github.com/Xilinx/linux-xlnx.git

In three different instances the return value of "of_property_read_u32"
API was neither captured nor checked.
Fixed it by capturing the return value and then checking for any error.

Addresses-Coverity: "check_return"
Signed-off-by: Srinivas Neeli <srinivas.neeli@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpio/gpio-xilinx.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/gpio/gpio-xilinx.c b/drivers/gpio/gpio-xilinx.c
index d94af260ea1d..38696f711154 100644
--- a/drivers/gpio/gpio-xilinx.c
+++ b/drivers/gpio/gpio-xilinx.c
@@ -612,18 +612,21 @@ static int xgpio_of_probe(struct platform_device *pdev)
 		return -ENOMEM;
 
 	/* Update GPIO state shadow register with default value */
-	of_property_read_u32(np, "xlnx,dout-default", &chip->gpio_state);
+	if (of_property_read_u32(np, "xlnx,dout-default", &chip->gpio_state))
+		dev_dbg(&pdev->dev, "Missing xlnx,dout-default property\n");
 
 	/* By default, all pins are inputs */
 	chip->gpio_dir = 0xFFFFFFFF;
 
 	/* Update GPIO direction shadow register with default value */
-	of_property_read_u32(np, "xlnx,tri-default", &chip->gpio_dir);
+	if (of_property_read_u32(np, "xlnx,tri-default", &chip->gpio_dir))
+		dev_dbg(&pdev->dev, "Missing xlnx,tri-default property\n");
 
 	chip->no_init = of_property_read_bool(np, "xlnx,no-init");
 
 	/* Update cells with gpio-cells value */
-	of_property_read_u32(np, "#gpio-cells", &cells);
+	if (of_property_read_u32(np, "#gpio-cells", &cells))
+		dev_dbg(&pdev->dev, "Missing gpio-cells property\n");
 
 	/*
 	 * Check device node and parent device node for device width
-- 
2.31.1

